docs/advanced.md

```diff
@@ -162,3 +162,61 @@ client = httpx.Client(proxies=proxy)
 # This request will be tunneled instead of forwarded.
 client.get("http://example.com")
 ```
+
+
+## Timeout fine-tuning
+HTTPX library offers many request timeout management options. HTTPX distinguish three types of timeouts: **connect** timeouts, 
+**write** timeouts and **read** timeouts.
+
+* Connect timeout specify time period within which a connection between a client and a server must be established.
+* Write timeout specify the maximum duration before timing out writes of the response.
+* Read timeout specify the maximum duration for reading the entire request.
+
+HTTPX allows you to manage timeout policy freely. You can set timeout parameter at client creation phase or pass it on every request.  
+In a nutshell, user can manage timeouts on three different levels of abstraction:
```

Comment:
It seems this part confuses how to define the *scope* of the timeout (client-wide vs. per-request) and how to define its *value* (e.g. as an integer vs. a `TimeoutConfig` instance)

I think we can split things up like this:

```markdown
### Setting timeouts

You can set timeouts on two levels:

- For a given request:

<!-- "Request level" example -->

- On a client instance, which results in the given `timeout` being used as a default for requests made with this client:

<!-- "Client level" example -->

Besides, you can pass timeouts in two forms:

- A number, which sets the read, write and connect timeouts to the same value, as in the examples above.
- A `TimeoutConfig` instance, which allows to define the read, write and connect timeouts independently:

<!-- Example -->
```